packages/ra-ui-materialui/src/list/FilterForm.js

```diff
@@ -61,71 +65,76 @@ const sanitizeRestProps = ({
     submitSucceeded,
     submitting,
     touch,
+    touched,
     triggerSubmit,
     untouch,
     valid,
     validate,
     validating,
-    _reduxForm,
+    values,
+    visited,
+    __versions,
     ...props
 }) => props;
 
-export class FilterForm extends Component {
-    componentDidMount() {
-        this.props.filters.forEach(filter => {
+export const FilterForm = ({
+    classes = {},
+    className,
+    resource,
+    margin,
+    variant,
+    filters,
+    displayedFilters,
+    hideFilter,
+    initialValues,
+    ...rest
+}) => {
+    useEffect(() => {
+        filters.forEach(filter => {
             if (filter.props.alwaysOn && filter.props.defaultValue) {
                 throw new Error(
                     'Cannot use alwaysOn and defaultValue on a filter input. Please set the filterDefaultValues props on the <List> element instead.'
                 );
             }
         });
-    }
-
-    getShownFilters() {
-        const { filters, displayedFilters, initialValues } = this.props;
+    }, [filters]);
```

Comment:
`filters` will change at each render, so this effect will run way too often. Maybe not significant though.